File: experiments/allocator/themain.py
from parlib.rtime import sleep
from parlib.directive import POSIX

from creator import Creator

def main(argv):
	n = 100000
	loops = 1000
	threads = 0
	if len(argv) >= 4:
		n = int(argv[1])
		loops = int(argv[2])
		threads = int(argv[3])
		
	c = Creator(n, loops)
	if threads == 1:
		c.start()
		sleep(1000.0)
	else:
		c.body()

if __name__ == "__main__":
	from sys import argv
	main(argv)

def entry_point(argv):
	if POSIX:
		from parlib.rthreading import init_threads
		init_threads()
	main(argv)
	return 0

def target(*argv):
	return entry_point, None

File: experiments/cppallocator/themain.cpp
/*
from parlib.rtime import sleep
from parlib.directive import POSIX

from creator import Creator

def main():
	c = Creator(100000)
	#c.start()
	#sleep(1000)
	c.body()

if __name__ == "__main__":
	main()

def entry_point(args):
	if POSIX:
		from parlib.rthreading import init_threads
		init_threads()
	main()
	return 0

def target(*args):
	return entry_point, None
*/

#include <cstdlib>
#include "creator.cpp"

int main(int argc, char* argv[]) {
	int n = 100000;
	int loops = 1000;
	int threads = 0;
	if (argc == 4) {
		n = atoi(argv[1]);
		loops = atoi(argv[2]);
		threads = atoi(argv[3]);
	}
	Creator * c = new Creator(n, loops);
	c->body();
	return 0;
}
